test/common/router/scoped_config_impl_test.cc

```diff
@@ -213,6 +216,13 @@ ScopeKey makeKey(const std::vector<const char*>& parts) {
   return key;
 }
 
+TEST(ScopeKeyDeathTest, AddNullFragment) {
+  ScopeKey key;
+#if !defined(NDEBUG)
+  EXPECT_DEATH(key.addFragment(nullptr), "null fragment not allowed in ScopeKey.");
```

Comment:
Use `EXPECT_DEBUG_DEATH` to avoid the `#if !defined(NDEBUG)` guard.

Reply:
done.